### DISCUSSION TITLE
Is there a linux version of wiresock?

### DISCUSSION THREAD

**Discussion by movsar** [wiresock/WireSockUI#62] [Q&A] [answered]
I'm using this app daily, this (using Socks5 for the handshake) is the only way I can connect to a VPN in Russia, but I'd like to create a VM as a router for a persistent connection and it would be nice to use a headless ubuntu for that, so, is there any way to use it there?

If not, do you know or have any idea how to set up WG connection through SOCKS5 on ubuntu? That is without sending all the traffic through Socks5

Thanks for this awesome project

**Comment by wiresock** [comment, marked as answer]
WireSock is exclusively available for Windows. However, since it is built on Cloudflare's BoringTun, it should be feasible to implement WireGuard handshake tunneling over SOCKS5 in the Linux version of the BoringTun application.

**Reply by movsar** [reply]
Thanks for the quick reply, I'll use windows 7 for now then :)
